### ISSUE TITLE
Providing descriptive captions for images on (educational/government) websites in India to  improve their accessibility

### ISSUE THREAD

**Issue by gayatrivenugopal** [datameet-pune/datameet-pune.github.io#11]
Accessibility refers the method of making a product easy to use for users irrespective of their
abilities. In case of web accessibility, the aim is to make it easy for users to read and understand
the content on the website. In this proposal, we focus on the accessibility of images to visually
impaired users. WCAG 2.0 are a set of guidelines by the World Wide Web that have been
published to make the content on websites more accessible to its consumers.
In order to improve the accessibility of images on websites, WCAG 2.0 provides different
solutions for different types of images. For instance, informative images such as photos should
be given a short description. Decorative images should not have any alternate text since their
purpose is only to make the page more attractive. Functional images are those that are displayed
on buttons and other controls that have associated actions. In such cases, the alternate text should
describe the action and not the image. Complex images such as graphs should be given elaborate
descriptions. Informative text should be avoided in images as much as possible. An image in a
group of images conveying the same meaning should have an alternative text such that it should
convey the meaning of all the images in the group. Image maps should have an alternative text
for each region.
Unfortunately many websites do not follow these guidelines which causes a negative effect on
the user experience, as they are unable to acquire the information that they were searching for.
Various tools can be found that have been created as a part of many studies carried out in the
area of image recognition. These tools perform either or all of the following:
1. Extracting text from images - Here, the focus is on images containing text. OCR software
solutions extract the text from the image. However, not many successful versions of OCR
could be found for Hindi.
2. Rendering the image without the text
One popular example is https://cloud.google.com/vision/ by Google. The tool accepts an image
and returns its auto-generated description. We tried to use this tool to read the following image:
![wsis-banner](https://user-images.githubusercontent.com/1294745/36363765-aa6ea1c2-1565-11e8-9229-f98faf028f66.jpg)

The generated text is given below::
Mess mmm oir REGISTER For Open Governemnt Data (0GD) Platform India -· CATEG0RY 3 ·
Last Date: 18th February 2018
Another site: https://egreetings.gov.in/, that is used to share e-greeting cards has a range of cards
to choose from but with no understandable alternate text. Two images of cards that fall in the
‘Holi greetings’ category are given below:

![1489123708_10-03-2017_4](https://user-images.githubusercontent.com/1294745/36363776-bfe7551c-1565-11e8-9112-ebf4f26b43cf.jpg)
![1489123708_10-03-2017_5](https://user-images.githubusercontent.com/1294745/36363777-c0529d18-1565-11e8-9559-26e9b0b88f61.jpg)

The alt text provided is ‘Holi | Greetings Portal’. Since images in such categories carry
cultural/religious/mythological graphics, it may not be possible for a generic image descriptor
tool to render the image and generate appropriate text; therefore manual intervention is required
in such scenarios.
To summarize, we propose to:
1. Select websites and identify the non-decorative images on all the pages of the site.
2. Categorize the images according to their type (as specified in WCAG 2.0).
3. Label each image with a tag (e.g. content, information, festival etc.).
4. Manually provide a suitable description in English/Hindi/other regional languages.
A repository of such images and their metadata could be created which could be used to map
with the websites they are available on.

References:
1. https://www.w3.org/TR/WCAG20/
2. https://www.w3.org/WAI/tutorials/images/
3. https://cloud.google.com/vision/

**Comment by answerquest** [comment]
Good docu! I suggest you start off a database. It can be as simple as a google form collecting image URLs and alt text advice. Target sites that visually challenged folks will be more likely to visit.

Programming idea: Imagine a browser extension that puts a "suggest an alternative text" option in the context menu when a user right-clicks on an image.

**Comment by gayatrivenugopal** [comment]
Thanks, the programming idea is good! Will start working on the image filtering part...

**Comment by AyushiGoel7200** [comment]
I'm attaching links to some targeting sites that visually challenged folks will be more likely to visit.
https://uidai.gov.in/
https://www.india.gov.in/
https://www.india.gov.in/my-government
https://indianvisaonline.gov.in/visa/index.html
https://www.incometaxindia.gov.in/Pages/default.aspx
https://www.irctc.co.in/eticketing/loginHome.jsf

**Comment by gayatrivenugopal** [comment]
Alternative to manual downloading of images: Extension to download all the images from the current page, along with the alt text, and save in a local compressed folder.
